Tweak some docs a bit (#311)
JuliaGaussianProcesses/KernelFunctions.jl: docs/src/userguide.md

```
This is the mechanism used throughout KernelFunctions.jl to handle multi-dimensional inputs.

You can utilise the `obsdim` keyword argument if you prefer:
```julia
kernelmatrix(k, X; obsdim=1) # same as RowVecs(X)
kernelmatrix(k, X; obsdim=2) # same as ColVecs(X)
```
This is similar to the convention used in [Distances.jl](https://github.com/JuliaStats/Distances.jl).

> ### So what type should I use to represent a collection of inputs?

Author: Here I tried to add some additional clarity to the previous bit where we gave users two options (ColVecs / RowVecs vs obsdim).

> The central assumption made by KernelFunctions.jl is that all collections of inputs of
> length `N` are represented by `AbstractVector`s of length `N`.

Reviewer: ```suggestion
The central assumption made by KernelFunctions.jl is that all collections of $N$ inputs are represented by `AbstractVector`s of length `N`.
```

Maybe this is less ambiguous (are the inputs of length N or the collection is of size N ?

Author: > are the inputs of length N or the collection is of size N

Both, that's the point :)

Reviewer: I meant it sounds like one input is a vector of length N

Author: Ohhhhhhhh sorry. I see.

Author: ```suggestion
The central assumption made by KernelFunctions.jl is that all collections of `N` inputs are represented by `AbstractVector`s of length `N`.
```

Abstraction is then used to ensure that efficiency is retained, `ColVecs` and `RowVecs`
being the most obvious examples of this.

Concretely:

> 1. For `Real`-valued inputs, a `Vector{<:Real}` is fine.

Reviewer: ```suggestion
1. For `Real`-valued inputs (scalars), a `Vector{<:Real}` is fine.
```

> 1. For vector-valued inputs, consider a `ColVecs` or `RowVecs`.
> 1. For a new input type, simply represent collections of inputs of this type as an `AbstractVector`.

Reviewer: ```suggestion
2. For vector-valued inputs, consider a `ColVecs` or `RowVecs`.
3. For a new input type, simply represent collections of inputs of this type as an `AbstractVector`.
```

Author: This is a markdown thing -- when the docs are built, it automatically replaces the 1s with appropriate numbers!
